Bukkit/src/main/java/com/plotsquared/bukkit/schematic/StateWrapper.java

```diff
@@ -44,25 +47,35 @@
 import org.bukkit.block.Skull;
 import org.bukkit.block.banner.Pattern;
 import org.bukkit.block.banner.PatternType;
+import org.bukkit.block.sign.Side;
+import org.bukkit.block.sign.SignSide;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.ItemStack;
 import org.checkerframework.checker.nullness.qual.NonNull;
+import org.jetbrains.annotations.ApiStatus;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Objects;
 import java.util.UUID;
+import java.util.function.BiConsumer;
 
+// TODO: somehow unbreak this class so it doesn't fuck up the whole schematic population system due to MC updates
```

Comment:
Probably want to add some kind of error recovery... potentially just use WE/FAWE adapters?

Reply:
Will have to see. I do have some work to do, specifically for signs as the message lines can be multiple arrays and compounds as well. The compounds should be piped through DataFixerUpper ig. Same for items in containers. 

But I don't think of any logging is really possible. I can't just log if the expected compounds don't exist - as they may simply not exist for other reasons. The easiest would obviously be piping all NBT that can be piped through DFU and use NMS to apply the state - but that's not really great either. I'm not really sure how to approach that in a safe, sane and future-proof way yet. But the fix should work as of now at least (or rather almost)

Reply:
good point - will look into it and see if it's possible to copy data to an arbitrary block state (reflection would be fine tbf)